src/Features/Core/Portable/DocumentationComments/CodeFixes/AbstractAddDocCommentNodesCodeFixProvider.cs

```diff
@@ -0,0 +1,166 @@
+        where TMemberDeclarationSyntax : SyntaxNode
+    {
+        public override FixAllProvider GetFixAllProvider() => WellKnownFixAllProviders.BatchFixer;
+
+        public async sealed override Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            var root = await context.Document.GetSyntaxRootAsync(context.CancellationToken).ConfigureAwait(false);
+            var parentMethod = root.FindNode(context.Span).FirstAncestorOrSelf<TMemberDeclarationSyntax>();
+
+            if (parentMethod != null)
+            {
+                context.RegisterCodeFix(
+                    new MyCodeAction(
+                        c => AddParamTagAsync(context.Document, context.Span, c)),
+                    context.Diagnostics);
+            }
+        }
+
+        protected abstract string NodeName { get; }
+
+        protected abstract List<TXmlNameAttributeSyntax> GetNameAttributes(TXmlElementSyntax node);
+        protected abstract string GetValueFromNameAttribute(TXmlNameAttributeSyntax attribute);
+        protected abstract SyntaxNode GetDocCommentNode(SyntaxTriviaList parameter);
+        protected abstract string GetXmlElementLocalName(TXmlElementSyntax element);
+        protected abstract List<string> GetParameterNames(TMemberDeclarationSyntax method);
+        protected abstract TXmlElementSyntax GetNewNode(string parameterName, bool isFirstNodeInComment);
+
+        protected async Task<Document> AddParamTagAsync(
+            Document document, TextSpan span, CancellationToken cancellationToken)
+        {
+            var root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+            var parentMethod = root.FindNode(span).FirstAncestorOrSelf<TMemberDeclarationSyntax>();
+            var docCommentNode = GetDocCommentNode(parentMethod.GetLeadingTrivia());
+
+            var newDocComment = docCommentNode;
+            var parameterNames = GetParameterNames(parentMethod);
+
+            foreach (var parameterName in parameterNames)
+            {
+                var paramNodes = GetNodes(newDocComment);
+                if (NodeExists(paramNodes, parameterName))
+                {
+                    continue;
+                }
+
+                var paramsBeforeCurrentParam = parameterNames.TakeWhile(t => t != parameterName).ToList();
+                var paramsAfterCurrentParam = parameterNames.Except(paramsBeforeCurrentParam).ToList();
+                paramsAfterCurrentParam.Remove(parameterName);
+
+                // First, try to get the node before the param node so we know where to insert the new node
+                SyntaxNode nodeBeforeNewParamNode = null;
+                if (paramsBeforeCurrentParam.Any())
+                {
+                    nodeBeforeNewParamNode = GetLastParamNodeCorrespondingToParamInList(paramNodes, paramsBeforeCurrentParam);
+                }
+
+                // This value is only useful if there are no nodes before the new node
```

Comment:
> This value is only useful if there are no nodes before the new node

So shouldn't there be a check in the 'if' for something like "if (!paramsBeforeCurrentParam.Any() && paramAfterCurrentParem.Any()"?